### PR TITLE
refactored file explorer

### PR DESCRIPTION
**Pull request by LianaHus**
- refactored file explorer and file panel

- fixed bug for empty file during create gist

- landing page restyleing

fixes https://github.com/ethereum/remix-ide/issues/1757
fixes https://github.com/ethereum/remix-ide/issues/1756

### CODE REVIEW

File: src/app/files/file-explorer.js [hunk 1]
```diff
@@ -276,6 +316,217 @@ fileExplorer.prototype.init = function () {
   return this.container
 }
 
+fileExplorer.prototype.publishToGist = function (fileProviderName) {
```
Comment (yann300) [inline src/app/files/file-explorer.js]: I think these functions don't not have any param anymore?

File: src/app/files/file-explorer.js [hunk 2]
```diff
@@ -31,6 +63,8 @@ function fileExplorer (localRegistry, files) {
     fileManager: self._components.registry.get('filemanager').api
   }
 
+  self._components.registry.put(`fileexplorer${files.type}`, this)
```
Comment (yann300) [inline src/app/files/file-explorer.js]: I don't think you need this anymore

File: src/app/files/file-explorer.js [hunk 3]
```diff
@@ -276,6 +316,217 @@ fileExplorer.prototype.init = function () {
...
 
+fileExplorer.prototype.publishToGist = function (fileProviderName) {
+  modalDialogCustom.confirm(
+    null,
+    'Are you sure you want to publish all your files anonymously as a public gist on github.com?',
+    () => { this.toGist(fileProviderName) }
+  )
+}
+
+fileExplorer.prototype.uploadFile = function (event) {
```
Comment (yann300) [inline src/app/files/file-explorer.js]: One thing that should perhaps be done in other PR is to move these functions to another file

**Comment by yann300** [comment]
browser tests are failing (I guess that because create new file action has moved)

File: src/app/files/file-explorer.js (hunk 1 above)
Comment (LianaHus) [reply src/app/files/file-explorer.js]: oh, forgot to remove this

**Pull request merged**